### PR TITLE
Improve types for using babel parser

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Tweaks type definitions for babel/parser

## Checklist

- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/parse/babel.js [hunk 1]
```diff
@@ -74,7 +85,7 @@ function isFlowFile(text, options) {
 
 function parseWithOptions(parseMethod, text, options) {
   // Inline the require to avoid loading all the JS if we don't use it
-  /** @type {import("@babel/parser").parse | import("@babel/parser").parseExpression} */
+  /** @type {Parse} */
```
Comment (fisker) [inline src/language-js/parse/babel.js:88]: Shouldn't it lowercased?

Comment (sosukesuzuki) [reply src/language-js/parse/babel.js:88]: I have no strong opinion. But, generally type names should be upper camel cased in TypeScript. `("@babel/parser").parse` and `import("@babel/parser").parseExpression` are function (not type) so they are lowercased.

Comment (sosukesuzuki) [reply src/language-js/parse/babel.js:88]: @fisker What do you think about the above comment?

Comment (fisker) [reply src/language-js/parse/babel.js:88]: I don't use jsdoc at all, I can't review.

**Pull request merged**